bokehjs/src/coffee/models/plots/plot.coffee

```diff
@@ -64,21 +66,46 @@ class Plot extends LayoutDOM.Model
   initialize: (options) ->
     super(options)
 
+    for xr in _.values(@extra_x_ranges).concat(@x_range)
+      plots = xr.get('plots')
```

Comment:
As a side note - `plots` doesn't seem like a great name for this property. It makes this code super confusing.